File: May-18/04-05-18/python/02-Level-node.py
N = 10**5 + 5
adj = {}
visit = [0]*N
level = [0]*N

def bfs(s):
    q = []
    q.append(s)
    visit[s] = 1
    level[s] = 1
    
    while(len(q) > 0):
        val = q[0]
        del q[0]
        
        for i in adj[val]:
            if visit[i] == 0:
                visit[i] = 1
                level[i] = level[val] + 1
                q.append(i)
                
    



n = int(input())

for i in range(1,n):
    x, y = map(int, input().split())
    tmp = []
    if x in adj:
        tmp = adj[x]
    tmp.append(y)
    adj[x] = tmp;
    
    tmp = []
    if y in  adj:
        tmp = adj[y]
    tmp.append(x)
    adj[y] = tmp

bfs(1)
l = int(input())
cnt = 0
for i in range(1, n):
    if(level[i] == l):
        cnt += 1
        
print(cnt)

File: May-18/04-05-18/java/02-Level-node.java

import java.util.*;

class TestClass {
    static int N = (int)1e5;
    static int level[] = new int[N];
    static boolean visit[] = new boolean[N];
    static LinkedList<Integer> adj[] = new LinkedList[N];
    
    static void bfs(int s)
    {
        LinkedList<Integer> q = new LinkedList<Integer>();
        q.add(s);
        visit[s] = true;
        level[s] = 1;
        
        while(q.size() != 0)
        {
            int v = q.poll();
            
            Iterator<Integer> i = adj[v].listIterator();
            while (i.hasNext())
            {
                int val = i.next();
                if (visit[val] == false)
                {
                    visit[val] = true;
                    q.add(val);
                    level[val] = level[v] + 1;
                }
            }
        }
    }
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        int n;
        n = input.nextInt();
        for(int i = 0; i <= n; i++)
            adj[i] = new LinkedList();
        for(int i = 1; i < n; i++)
        {
            int x, y;
            x = input.nextInt();
            y = input.nextInt();
            adj[x].add(y);
            adj[y].add(x);
        }
        bfs(1);
        int l = input.nextInt();
        int cnt = 0;
        for(int i = 1; i < n; i++)
        {
            if(level[i] == l)
                cnt += 1;
        }
        System.out.println(cnt);
        
    }
}
